packages/react-core/src/next/components/Modal/ModalBoxTitle.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import modalStyles from '@patternfly/react-styles/css/components/ModalBox/modal-box';
+import { css } from '@patternfly/react-styles';
+import { capitalize } from '../../../helpers';
+import { Tooltip } from '../../../components/Tooltip';
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import InfoCircleIcon from '@patternfly/react-icons/dist/esm/icons/info-circle-icon';
+import BellIcon from '@patternfly/react-icons/dist/esm/icons/bell-icon';
+import { useIsomorphicLayoutEffect } from '../../../helpers';
+
+export const isVariantIcon = (icon: any): icon is string =>
+  ['success', 'danger', 'warning', 'info', 'custom'].includes(icon as string);
+
+export interface ModalBoxTitleProps {
+  /** Additional classes added to the modal box title. */
+  className?: string;
+  /** Id of the modal box title. */
+  id?: string;
+  /** Content rendered inside the modal box title. */
+  title: React.ReactNode;
+  /** Optional alert icon (or other) to show before the title. When the predefined alert types
+   * are used the default styling will be automatically applied. */
+  titleIconVariant?: 'success' | 'danger' | 'warning' | 'info' | 'custom' | React.ComponentType<any>;
+  /** Optional title label text for screen readers. */
+  titleLabel?: string;
```

Comment:
This could probably be changed to `titleScreenReaderText` to match similar props in other components.